pulsar-client/src/main/java/org/apache/pulsar/client/impl/conf/ConsumerConfigurationData.java

```diff
@@ -90,6 +90,9 @@
 
     private DeadLetterPolicy deadLetterPolicy;
 
+    private boolean trackPartitionUpdate = false;
```

Comment:
I'd say to enable by default. The update is lightweight (it just needs to fetch a number from broker) so it would not be causing additional load.